web/packages/teleterm/src/services/config/getKeyboardShortcutSchema.ts

```diff
@@ -0,0 +1,175 @@
+  'Valid shortcut can contain multiple modifiers (like "Shift") and a single key code (like "A" or "Tab"), combined by the "+" character.';
+
+export function invalidKeyCodeIssue(wrongKeyCode: string): z.IssueData {
+  return {
+    code: z.ZodIssueCode.custom,
+    message: `"${wrongKeyCode}" cannot be used as a key code. ${VALID_SHORTCUT_MESSAGE}`,
+  };
+}
+
+export function invalidModifierIssue(wrongModifier: string): z.IssueData {
+  return {
+    code: z.ZodIssueCode.custom,
+    message: `"${wrongModifier}" cannot be used as a modifier. ${VALID_SHORTCUT_MESSAGE}`,
+  };
+}
+
+export function duplicatedModifierIssue(): z.IssueData {
+  return {
+    code: z.ZodIssueCode.custom,
+    message: `Duplicated modifier found. ${VALID_SHORTCUT_MESSAGE}`,
+  };
+}
+
+export function getKeyboardShortcutSchema(platform: Platform) {
+  const allowedModifiers = getAllowedModifiers(platform);
+
+  return z
+    .string()
+    .transform(s => s.split('+'))
+    .transform(putModifiersFirst(allowedModifiers))
+    .superRefine(validateKeyCodeAndModifiers(allowedModifiers))
+    .transform(s => s.join('+'));
+}
+
+function putModifiersFirst(
+  allowedModifiers: string[]
+): (tokens: string[]) => string[] {
+  return tokens =>
+    tokens.sort((a, b) => {
+      if (allowedModifiers.indexOf(a) === -1) {
+        return 1;
+      }
+      if (allowedModifiers.indexOf(b) === -1) {
+        return -1;
+      }
+      return allowedModifiers.indexOf(a) - allowedModifiers.indexOf(b);
+    });
+}
+
+function validateKeyCodeAndModifiers(
+  allowedModifiers: string[]
+): (tokens: string[], ctx: z.RefinementCtx) => void {
+  return (tokens, ctx) => {
+    // empty accelerator disables the shortcut
+    if (tokens.join('') === '') {
+      return z.NEVER;
+    }
+
+    const [expectedKeyCode, ...expectedModifiers] = [...tokens].reverse();
+
+    if (!ALLOWED_KEY_CODES.includes(expectedKeyCode)) {
+      ctx.addIssue(invalidKeyCodeIssue(expectedKeyCode));
+      return z.NEVER;
+    }
+
+    if (expectedModifiers.length !== new Set(expectedModifiers).size) {
+      ctx.addIssue(duplicatedModifierIssue());
+      return z.NEVER;
+    }
+
+    expectedModifiers.forEach(expectedModifier => {
+      if (!allowedModifiers.includes(expectedModifier)) {
+        ctx.addIssue(invalidModifierIssue(expectedModifier));
+      }
+    });
+  };
+}
+
+/** Returns allowed modifiers for a given platform in a correct order. */
+function getAllowedModifiers(platform: Platform): string[] {
+  switch (platform) {
+    case 'win32':
+    case 'linux':
+      return ['Ctrl', 'Alt', 'Shift'];
+    case 'darwin':
+      return ['Command', 'Control', 'Option', 'Shift'];
+  }
+}
+
+function generateRange(start: number, end: number): string[] {
+  return new Array(end - start + 1)
+    .fill(undefined)
+    .map((_, i) => (i + start).toString());
+}
+
+const ALLOWED_KEY_CODES = [
+  ...generateRange(0, 9), // 0-9 range
+  ...generateRange(1, 24).map(key => `F${key}`), // F1-F24 keys
```

Comment:
I added it because I saw this post https://dev.to/grahamthedev/did-you-know-there-are-f13-f24-keys-368p, but I think F13-24 are not standard (?), so perhaps I will remove them.

Reply:
> I added it because I saw this post https://dev.to/grahamthedev/did-you-know-there-are-f13-f24-keys-368p, but I think F13-24 are not standard (?), so perhaps I will remove them.

Why though? Electron supports them OOTB. The post mentions Stream Deck which was also a subject of a discussion on one of the Slack channels this week. ;)